iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/iothub/TokenRenewalTests.java

```diff
@@ -75,9 +75,9 @@ public static void stopProxy()
     @Test
     public void tokenRenewalWorks() throws Exception
     {
-        final long SECONDS_FOR_SAS_TOKEN_TO_LIVE_BEFORE_RENEWAL = 60;
-        final long EXPIRED_SAS_TOKEN_GRACE_PERIOD_SECONDS = 600;
-        final long EXTRA_BUFFER_TO_ENSURE_TOKEN_EXPIRED_SECONDS = 30;
+        final long SECONDS_FOR_SAS_TOKEN_TO_LIVE_BEFORE_RENEWAL = 30;
+        final long EXPIRED_SAS_TOKEN_GRACE_PERIOD_SECONDS = 600; //service extends 10 minute grace period after a token has expired
+        final long EXTRA_BUFFER_TO_ENSURE_TOKEN_EXPIRED_SECONDS = 120; //wait time beyond the expected grace period, just in case
```

Comment:
so we need 2 extra minutes for the test to succeed?

Reply:
Yes, this test needs to open a connection, let the first token expire, and then let the service grace period expire. Only after all of that will the test check things like that the MQTT connection was lost.